src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java

```diff
@@ -74,4 +76,27 @@ Page<EventSearchResult> searchEventsFilteringOn(
         LocalDate hearingStartDate,
         LocalDate hearingEndDate,
         Pageable pageable);
-}
+
+    @Query(value = """
+            select distinct evnt.event_id from darts.event evnt
+                inner join (SELECT distinct count(event_id) as eventCount, event_id  FROM
+                darts.event where is_current=true GROUP BY event_id) eventIdCntResult ON evnt.event_id = eventIdCntResult.event_id
+            where evnt.event_id <> 0 and eventIdCntResult.eventCount > 1
+            LIMIT :batchSize
+        """, nativeQuery = true)
+    List<Integer> getCurrentEventIdsToBeProcessed(int batchSize);
+
+    @Query(value = """
+             select eve_id FROM darts.event e where event_id=:eventId order by created_ts desc LIMIT 1
+        """, nativeQuery = true)
+    Integer getTheCurrentEventPrimaryKeyForEventId(int eventId);
```

Comment:
```suggestion
    Integer getLatestEventPrimaryKeyForEventId(int eventId);
```

Reply:
Was trying to use the terminology of the database. This is the primary key of the current event